push-build.sh

```diff
@@ -117,34 +117,33 @@ RELEASE_BUCKET=${FLAGS_bucket:-"kubernetes-release-dev"}
 [[ $KUBE_GCS_UPDATE_LATEST == "n" ]] && FLAGS_noupdatelatest=1
 
 # Default to kubernetes
-: ${FLAGS_release_kind:="kubernetes"}
+: "${FLAGS_release_kind:="kubernetes"}"
 
 # This will canonicalize the path
 KUBE_ROOT=$(pwd -P)
 
 USE_BAZEL=false
-if release::was_built_with_bazel $KUBE_ROOT $FLAGS_release_kind; then
+if release::was_built_with_bazel "$KUBE_ROOT" $FLAGS_release_kind; then
   USE_BAZEL=true
   bazel build //:version
-  LATEST=$(cat $KUBE_ROOT/bazel-genfiles/version)
+  LATEST=$(cat "$KUBE_ROOT"/bazel-genfiles/version)
 else
-  LATEST=$(tar -O -xzf $KUBE_ROOT/_output/release-tars/$FLAGS_release_kind.tar.gz $FLAGS_release_kind/version)
+  LATEST=$(tar -O -xzf "$KUBE_ROOT"/_output/release-tars/$FLAGS_release_kind.tar.gz $FLAGS_release_kind/version)
 fi
-
 if [[ "$LATEST" =~ (${VER_REGEX[release]}(\.${VER_REGEX[build]})?(-dirty)?) ]]; then
-  if ((FLAGS_ci)) && [[ "$LATEST" =~ "dirty" ]]; then
+  if ((${FLAGS_ci:?})) && [[ "$LATEST" =~ "dirty" ]]; then
```

Comment:
so all of these `FLAGS_.*` environment variables are magically generated based on flags being present, I.E> if `--ci` is passed we get `FLAGS_ci`. They're not set when the flag isn't passed.

checking them with `:?` is going to break this script.

Reply:
I'm not sure what the best approach to dealing with these is though, the whole system is a bit magical 🤔

Reply:
Kubernetes is magical :)